File: apps/ledger-live-mobile/ios/AppleWalletModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AppleWalletModule, NSObject)

RCT_EXTERN_METHOD(openPaymentSetup:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: apps/ledger-live-mobile/ios/AppleWalletModule.swift
import PassKit
import React

@objc(AppleWalletModule)
final class AppleWalletModule: NSObject, RCTBridgeModule {
  static func moduleName() -> String! { "AppleWalletModule" }
  static func requiresMainQueueSetup() -> Bool { false }

  @objc(openPaymentSetup:rejecter:)
  func openPaymentSetup(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    DispatchQueue.main.async {
      guard PKPaymentAuthorizationController.canMakePayments() else {
        reject("apple_pay_unavailable", "Apple Pay is not available on this device", nil)
        return
      }

      PKPassLibrary().openPaymentSetup()
      resolve(nil)
    }
  }
}
